nni/contrib/compression/utils/evaluator.py

```diff
@@ -185,6 +186,51 @@ def unbind_model(self):
         """
         raise NotImplementedError
 
+    def _optimizer_add_param_group(self, model: Union[torch.nn.Module, pl.LightningModule],
+                                  module_name_param_dict: Dict[str, List[Tensor]], optimizers: Optimizer | List[Optimizer]):
+        # used in the bind_model process
+        def find_param_group(param_groups: List[Dict], module_name: str):
+            for param_group in param_groups:
+                params = param_group["params"]
+                if isinstance(params, Tensor):
+                    params = [params]
+                elif isinstance(params, set):
+                    raise TypeError('optimizer parameters need to be organized in ordered collections, but '
+                                    'the ordering of tensors in sets will change between runs. Please use a list instead.')
+                else:
+                    params = list(params)
+                name_lis = [param2name_dict[id(p)] for p in params]
+
+                for name in name_lis:
+                    if module_name in name:
+                        return param_group
+
+            return None
+
+        def add_param(param_lis: List[Tensor], target_param_group: Dict, optimizer: Optimizer):
+            for param in param_lis:
+                new_param_group = target_param_group.copy()
```

Comment:
copy() is a shadow copy, is this enough? or will some other key-value pairs be related to `params`?

Reply:
after check optim algorithms in the torch.optim, I find that types of all values in the param_group are  Union[str, float, bool, Tuple[float, float]], so I think copy() is enough.